In [1]:
# ライブラリのインポート
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
# 元となるデータの取得
df_origin = pd.read_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-6/user-6-nonull.csv")
df_origin.head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
0           0                    0           843115863          NaN   
1           1                12473           843128336        Mouse   
2           2                18888           843134751  Eye Tracker   
3           3                27002           843142865        Mouse   
4           4                43069           843158932        Mouse   

  Project name Export date Participant name Recording name Recording date  \
0          JD4  2025/07/18           user-6     Recording7     2025/07/18   
1          JD4  2025/07/18           user-6     Recording7     2025/07/18   
2          JD4  2025/07/18           user-6     Recording7     2025/07/18   
3          JD4  2025/07/18           user-6     Recording7     2025/07/18   
4          JD4  2025/07/18           user-6     Recording7     2025/07/18   

  Recording date UTC  ... Original Media height Eye movement type  \
0         2025/07/18  ...                   NaN               NaN   
1         2025/07/18  ...                   NaN      EyesNotFound   
2         2025/07/18  ...                   NaN      EyesNotFound   
3         2025/07/18  ...                   NaN      EyesNotFound   
4         2025/07/18  ...                   NaN      EyesNotFound   

   Eye movement event duration Eye movement type index Fixation point X  \
0                          NaN                     NaN              NaN   
1                         76.0                     1.0              NaN   
2                         76.0                     1.0              NaN   
3                         76.0                     1.0              NaN   
4                         76.0                     1.0              NaN   

  Fixation point Y  Fixation point X (MCSnorm)  Fixation point Y (MCSnorm)  \
0              NaN                         NaN                         NaN   
1              NaN                         NaN                         NaN   
2              NaN                         NaN                         NaN   
3              NaN                         NaN                         NaN   
4              NaN                         NaN                         NaN   

   Mouse position X  Mouse position Y  
0               NaN               NaN  
1            1809.0             777.0  
2               NaN               NaN  
3            1809.0             759.0  
4            1809.0             746.0  

[5 rows x 84 columns]

# カラムの整理

In [3]:
nunique_series = df_origin.nunique(dropna=False).to_string()
print(nunique_series)

Unnamed: 0                                    11021
Recording timestamp                           11008
Computer timestamp                            11008
Sensor                                            3
Project name                                      1
Export date                                       1
Participant name                                  1
Recording name                                    1
Recording date                                    1
Recording date UTC                                1
Recording start time                              1
Recording start time UTC                          1
Recording duration                                1
Timeline name                                     1
Recording Fixation filter name                    1
Recording software version                        1
Recording resolution height                       1
Recording resolution width                        1
Recording monitor latency                         1
Average cali

In [4]:
# 欠損値も含めてユニークな値の個数をカウント
nunique_series = df_origin.nunique(dropna=False)

# ユニークな値が1つのカラム名リスト
single_unique_cols = nunique_series[nunique_series == 1].index.tolist()

In [5]:
# カラム名とそのユニーク値を出力
for col in single_unique_cols:
    unique_val = df_origin[col].unique()
    print(f"{col}: {unique_val}")


Project name: ['JD4']
Export date: ['2025/07/18']
Participant name: ['user-6']
Recording name: ['Recording7']
Recording date: ['2025/07/18']
Recording date UTC: ['2025/07/18']
Recording start time: ['16:43:00.114']
Recording start time UTC: ['07:43:00.114']
Recording duration: [198601]
Timeline name: ['Timeline1']
Recording Fixation filter name: ['Tobii I-VT (Fixation)']
Recording software version: ['24.21.435']
Recording resolution height: [1080]
Recording resolution width: [1920]
Recording monitor latency: [10]
Average calibration accuracy (mm): [11.9]
Average calibration accuracy (degrees): [0.87]
Average calibration accuracy (pixels): [41]
Average validation accuracy (mm): [11.6]
Average validation precision SD (mm): [8.5]
Average validation precision RMS (mm): [10.5]
Average validation accuracy (degrees): [0.87]
Average validation precision SD (degrees): [0.64]
Average validation precision RMS (degrees): [0.79]
Average validation accuracy (pixels): [40]
Average validation precisio

## カラムの要素が1つしかない変数の説明
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-6`|
|Recording name| 録画名|`Recording7`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`16:43:00.114`|
|Recording start time UTC| 録画開始時刻(UTC) |`07:43:00.114`|
|Recording duration| 録画時間|`198601`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`11.9`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`0.87`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`41`pixels|
|Average validation accuracy | 平均バリデーション精度|`11.6`mm|
|Average validation precision SD | 平均バリデーション予測SD|`8.5`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`10.5`mm|
|Average validation accuracy |平均バリデーション精度| `0.87`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `0.64`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `0.79`degrees|
|Average validation accuracy |平均バリデーション精度| `40`pixels|
|Average validation precision SD |平均バリデーション予測SD| `30`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `37`pixels|


In [6]:
# 削除した新しいデータフレーム
df_cleaned = df_origin.drop(columns=single_unique_cols)
df_cleaned.head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
0           0                    0           843115863          NaN   
1           1                12473           843128336        Mouse   
2           2                18888           843134751  Eye Tracker   
3           3                27002           843142865        Mouse   
4           4                43069           843158932        Mouse   

   Eyetracker timestamp           Event Event value  Gaze point X  \
0                   NaN  RecordingStart         NaN           NaN   
1                   NaN             NaN         NaN           NaN   
2           858482516.0             NaN         NaN           NaN   
3                   NaN             NaN         NaN           NaN   
4                   NaN             NaN         NaN           NaN   

   Gaze point Y  Gaze point left X  ...  Original Media height  \
0           NaN                NaN  ...                    NaN   
1           NaN                NaN  ...                    NaN   
2           NaN                NaN  ...                    NaN   
3           NaN                NaN  ...                    NaN   
4           NaN                NaN  ...                    NaN   

   Eye movement type  Eye movement event duration  Eye movement type index  \
0                NaN                          NaN                      NaN   
1       EyesNotFound                         76.0                      1.0   
2       EyesNotFound                         76.0                      1.0   
3       EyesNotFound                         76.0                      1.0   
4       EyesNotFound                         76.0                      1.0   

   Fixation point X  Fixation point Y  Fixation point X (MCSnorm)  \
0               NaN               NaN                         NaN   
1               NaN               NaN                         NaN   
2               NaN               NaN                         NaN   
3               NaN               NaN                         NaN   
4               NaN               NaN                         NaN   

   Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
0                         NaN               NaN               NaN  
1                         NaN            1809.0             777.0  
2                         NaN               NaN               NaN  
3                         NaN            1809.0             759.0  
4                         NaN            1809.0             746.0  

[5 rows x 57 columns]

## ユニークな値を取り除いたデータフレーム

In [7]:
print(df_cleaned.nunique(dropna=False).to_string())

Unnamed: 0                             11021
Recording timestamp                    11008
Computer timestamp                     11008
Sensor                                     3
Eyetracker timestamp                   10797
Event                                      9
Event value                                9
Gaze point X                            1725
Gaze point Y                            1028
Gaze point left X                       1742
Gaze point left Y                       1057
Gaze point right X                      1748
Gaze point right Y                      1020
Gaze direction left X                   9523
Gaze direction left Y                   8750
Gaze direction left Z                   4293
Gaze direction right X                  9334
Gaze direction right Y                  8584
Gaze direction right Z                  4303
Pupil diameter left                      936
Pupil diameter right                     961
Pupil diameter filtered                  910
Validity l

## Sensor事にデータフレームを分割

In [8]:
# NaNを含む行
df_sensor_nan = df_cleaned[df_cleaned['Sensor'].isna()]

# "Mouse" の行
df_sensor_mouse = df_cleaned[df_cleaned['Sensor'] == "Mouse"]

# "Eye Tracker" の行
df_sensor_eye = df_cleaned[df_cleaned['Sensor'] == "Eye Tracker"]


### 問題の既知代わりの時間を取得する

In [9]:
df_sensor_nan

Unnamed: 0  Recording timestamp  Computer timestamp Sensor  \
0               0                    0           843115863    NaN   
184           184              2890231           846006094    NaN   
364           364              5123702           848239565    NaN   
373           373              5237915           848353778    NaN   
1792         1792             45422056           888537919    NaN   
1798         1798             45546008           888661871    NaN   
1820         1820             45997023           889112886    NaN   
1821         1821             45997023           889112886    NaN   
2132         2132             51021699           894137562    NaN   
2133         2133             51021699           894137562    NaN   
3902         3902             80385741           923501604    NaN   
3905         3905             80417213           923533076    NaN   
3906         3906             80417213           923533076    NaN   
4208         4208             85424828           928540691    NaN   
4209         4209             85424828           928540691    NaN   
6131         6131            117320571           960436434    NaN   
6134         6134            117352761           960468624    NaN   
6135         6135            117352761           960468624    NaN   
6437         6437            122360921           965476784    NaN   
6438         6438            122360921           965476784    NaN   
8417         8417            155290734           998406597    NaN   
8421         8421            155328191           998444054    NaN   
8422         8422            155328191           998444054    NaN   
8723         8723            160319323          1003435186    NaN   
8724         8724            160319323          1003435186    NaN   
10711       10711            193374455          1036490318    NaN   
10714       10714            193404009          1036519872    NaN   
10715       10715            193404009          1036519872    NaN   
11017       11017            198411737          1041527600    NaN   
11020       11020            198601390          1041717253    NaN   

       Eyetracker timestamp                          Event Event value  \
0                       NaN                 RecordingStart         NaN   
184                     NaN  Eye tracker Calibration start         NaN   
364                     NaN                     MouseEvent  Down, Left   
373                     NaN                     MouseEvent    Up, Left   
1792                    NaN                     MouseEvent  Down, Left   
1798                    NaN                     MouseEvent    Up, Left   
1820                    NaN    Eye tracker Calibration end         NaN   
1821                    NaN             ImageStimulusStart      blank2   
2132                    NaN               ImageStimulusEnd      blank2   
2133                    NaN             ImageStimulusStart  視線誘導問題.001   
3902                    NaN                  KeyboardEvent       Space   
3905                    NaN               ImageStimulusEnd  視線誘導問題.001   
3906                    NaN             ImageStimulusStart      blank2   
4208                    NaN               ImageStimulusEnd      blank2   
4209                    NaN             ImageStimulusStart  視線誘導問題.003   
6131                    NaN                  KeyboardEvent       Space   
6134                    NaN               ImageStimulusEnd  視線誘導問題.003   
6135                    NaN             ImageStimulusStart      blank2   
6437                    NaN               ImageStimulusEnd      blank2   
6438                    NaN             ImageStimulusStart  視線誘導問題.004   
8417                    NaN                  KeyboardEvent       Space   
8421                    NaN               ImageStimulusEnd  視線誘導問題.004   
8422                    NaN             ImageStimulusStart      blank2   
8723                    NaN               ImageStimulusEnd      blank2   
8724             

In [10]:
# "視線誘導問題." を含む行を抽出
df_filtered = df_sensor_nan[df_sensor_nan["Event value"].astype(str).str.contains("視線誘導問題.")].copy()

# 結果の確認（先頭5行を表示）
df_filtered.loc[:,['Recording timestamp', 'Computer timestamp', 'Event value']]
print(df_filtered.loc[:,['Recording timestamp', 'Computer timestamp', 'Event value']])

       Recording timestamp  Computer timestamp Event value
2133              51021699           894137562  視線誘導問題.001
3905              80417213           923533076  視線誘導問題.001
4209              85424828           928540691  視線誘導問題.003
6134             117352761           960468624  視線誘導問題.003
6438             122360921           965476784  視線誘導問題.004
8421             155328191           998444054  視線誘導問題.004
8724             160319323          1003435186  視線誘導問題.002
10714            193404009          1036519872  視線誘導問題.002


### 各問題の開始時刻
|       |Recording timestamp|  Computer timestamp| Event value|
|-------|-------------------|--------------------|------------|
|2133|51021699|894137562|視線誘導問題.001|
|3905|80417213|923533076|視線誘導問題.001|
|4209|85424828|928540691|視線誘導問題.003|
|6134|117352761|960468624|視線誘導問題.003|
|6438|122360921|965476784|視線誘導問題.004|
|8421|155328191|998444054|視線誘導問題.004|
|8724|160319323|1003435186|視線誘導問題.002|
|10714|193404009|1036519872|視線誘導問題.002|

In [11]:
df_filtered.loc[:,['Recording timestamp', 'Computer timestamp', 'Event value']].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-6/start-stop-timing.csv")

In [12]:
df_sensor_mouse

Unnamed: 0  Recording timestamp  Computer timestamp Sensor  \
1              1                12473           843128336  Mouse   
3              3                27002           843142865  Mouse   
4              4                43069           843158932  Mouse   
5              5                58870           843174733  Mouse   
7              7                74949           843190812  Mouse   
...          ...                  ...                 ...    ...   
3748        3748             77839572           920955435  Mouse   
6120        6120            117202003           960317866  Mouse   
6121        6121            117215921           960331784  Mouse   
6127        6127            117292677           960408540  Mouse   
6128        6128            117299687           960415550  Mouse   

      Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
1                      NaN   NaN         NaN           NaN           NaN   
3                      NaN   NaN         NaN           NaN           NaN   
4                      NaN   NaN         NaN           NaN           NaN   
5                      NaN   NaN         NaN           NaN           NaN   
7                      NaN   NaN         NaN           NaN           NaN   
...                    ...   ...         ...           ...           ...   
3748                   NaN   NaN         NaN           NaN           NaN   
6120                   NaN   NaN         NaN           NaN           NaN   
6121                   NaN   NaN         NaN           NaN           NaN   
6127                   NaN   NaN         NaN           NaN           NaN   
6128                   NaN   NaN         NaN           NaN           NaN   

      Gaze point left X  ...  Original Media height  Eye movement type  \
1                   NaN  ...                    NaN       EyesNotFound   
3                   NaN  ...                    NaN       EyesNotFound   
4                   NaN  ...                    NaN       EyesNotFound   
5                   NaN  ...                    NaN       EyesNotFound   
7                   NaN  ...                    NaN       EyesNotFound   
...                 ...  ...                    ...                ...   
3748                NaN  ...                 1080.0           Fixation   
6120                NaN  ...                 1080.0           Fixation   
6121                NaN  ...                 1080.0            Saccade   
6127                NaN  ...                 1080.0           Fixation   
6128                NaN  ...                 1080.0           Fixation   

      Eye movement event duration  Eye movement type index  Fixation point X  \
1                            76.0                      1.0               NaN   
3                            76.0                      1.0               NaN   
4                            76.0                      1.0               NaN   
5                            76.0                      1.0               NaN   
7                            76.0                      1.0               NaN   
...                           ...                      ...               ...   
3748                        399.0                    157.0             800.0   
6120                        266.0                    290.0            1139.0   
6121                         33.0                    342.0               NaN   
6127                        432.0                    291.0            1095.0   
6128                        432.0                    291.0            1095.0   

      Fixation point Y  Fixation point X (MCSnorm)  \
1                  NaN                         NaN   
3                  NaN                         NaN   
4                  NaN                         NaN   
5                  NaN                         NaN   
7                  NaN                         NaN   
...                ...                         ...   
3748             552.0                      0.41

In [13]:
df_sensor_eye

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
2               2                18888           843134751  Eye Tracker   
6               6                74431           843190294  Eye Tracker   
9               9                91074           843206937  Eye Tracker   
11             11               107761           843223624  Eye Tracker   
15             15               155926           843271789  Eye Tracker   
...           ...                  ...                 ...          ...   
11014       11014            198376477          1041492340  Eye Tracker   
11015       11015            198392520          1041508383  Eye Tracker   
11016       11016            198409704          1041525567  Eye Tracker   
11018       11018            198426396          1041542259  Eye Tracker   
11019       11019            198442443          1041558306  Eye Tracker   

       Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
2              8.584825e+08   NaN         NaN           NaN           NaN   
6              8.585381e+08   NaN         NaN           NaN           NaN   
9              8.585547e+08   NaN         NaN        1554.0        1004.0   
11             8.585714e+08   NaN         NaN           NaN           NaN   
15             8.586195e+08   NaN         NaN           NaN           NaN   
...                     ...   ...         ...           ...           ...   
11014          1.056828e+09   NaN         NaN         729.0         776.0   
11015          1.056844e+09   NaN         NaN         588.0        1003.0   
11016          1.056861e+09   NaN         NaN         550.0        1032.0   
11018          1.056878e+09   NaN         NaN         527.0        1034.0   
11019          1.056894e+09   NaN         NaN         528.0        1023.0   

       Gaze point left X  ...  Original Media height  Eye movement type  \
2                    NaN  ...                    NaN       EyesNotFound   
6                    NaN  ...                    NaN       EyesNotFound   
9                 1405.0  ...                    NaN       Unclassified   
11                   NaN  ...                    NaN       EyesNotFound   
15                   NaN  ...                    NaN       EyesNotFound   
...                  ...  ...                    ...                ...   
11014              714.0  ...                 1080.0            Saccade   
11015              604.0  ...                 1080.0            Saccade   
11016              561.0  ...                 1080.0            Saccade   
11018              579.0  ...                    NaN       Unclassified   
11019              559.0  ...                    NaN       Unclassified   

       Eye movement event duration  Eye movement type index  Fixation point X  \
2                             76.0                      1.0               NaN   
6                             76.0                      1.0               NaN   
9                             17.0                      1.0               NaN   
11                           171.0                      2.0               NaN   
15                           171.0                      2.0               NaN   
...                            ...                      ...               ...   
11014                         50.0                    654.0               NaN   
11015                         50.0                    654.0               NaN   
11016                         50.0                    654.0               NaN   
11018                         36.0                    104.0               NaN   
11019                         36.0                    104.0               NaN   

       Fixation point Y  Fixation point X (MCSnorm)  \
2                   NaN                         NaN   
6                   NaN                         NaN   
9                   NaN                         NaN   
11                  NaN                         NaN   
15                  NaN         

## Mouse以外のデータを抽出

In [14]:
df_cleaned_ver2 = df_cleaned[df_cleaned['Sensor'] != "Mouse"]
nunique_series_ver2 = df_cleaned_ver2.nunique(dropna=False)
single_unique_cols_ver2 = nunique_series_ver2[nunique_series_ver2 == 1].index.tolist()
# カラム名とそのユニーク値を出力
for col in single_unique_cols_ver2:
    unique_val = df_cleaned_ver2[col].unique()
    print(f"{col}: {unique_val}")

Mouse position X: [nan]
Mouse position Y: [nan]


In [15]:
df_cleaned_ver2 = df_cleaned_ver2.drop(columns=single_unique_cols_ver2)

In [16]:
print(df_cleaned_ver2.nunique(dropna=False).to_string())

Unnamed: 0                             10826
Recording timestamp                    10817
Computer timestamp                     10817
Sensor                                     2
Eyetracker timestamp                   10797
Event                                      9
Event value                                9
Gaze point X                            1725
Gaze point Y                            1028
Gaze point left X                       1742
Gaze point left Y                       1057
Gaze point right X                      1748
Gaze point right Y                      1020
Gaze direction left X                   9523
Gaze direction left Y                   8750
Gaze direction left Z                   4293
Gaze direction right X                  9334
Gaze direction right Y                  8584
Gaze direction right Z                  4303
Pupil diameter left                      936
Pupil diameter right                     961
Pupil diameter filtered                  910
Validity l

In [17]:
df_cleaned_ver2.head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
0            0                    0           843115863          NaN   
2            2                18888           843134751  Eye Tracker   
6            6                74431           843190294  Eye Tracker   
9            9                91074           843206937  Eye Tracker   
11          11               107761           843223624  Eye Tracker   

    Eyetracker timestamp           Event Event value  Gaze point X  \
0                    NaN  RecordingStart         NaN           NaN   
2            858482516.0             NaN         NaN           NaN   
6            858538056.0             NaN         NaN           NaN   
9            858554698.0             NaN         NaN        1554.0   
11           858571384.0             NaN         NaN           NaN   

    Gaze point Y  Gaze point left X  ...  Presented Media position Y (DACSpx)  \
0            NaN                NaN  ...                                  NaN   
2            NaN                NaN  ...                                  NaN   
6            NaN                NaN  ...                                  NaN   
9         1004.0             1405.0  ...                                  NaN   
11           NaN                NaN  ...                                  NaN   

    Original Media width  Original Media height  Eye movement type  \
0                    NaN                    NaN                NaN   
2                    NaN                    NaN       EyesNotFound   
6                    NaN                    NaN       EyesNotFound   
9                    NaN                    NaN       Unclassified   
11                   NaN                    NaN       EyesNotFound   

    Eye movement event duration  Eye movement type index  Fixation point X  \
0                           NaN                      NaN               NaN   
2                          76.0                      1.0               NaN   
6                          76.0                      1.0               NaN   
9                          17.0                      1.0               NaN   
11                        171.0                      2.0               NaN   

    Fixation point Y  Fixation point X (MCSnorm)  Fixation point Y (MCSnorm)  
0                NaN                         NaN                         NaN  
2                NaN                         NaN                         NaN  
6                NaN                         NaN                         NaN  
9                NaN                         NaN                         NaN  
11               NaN                         NaN                         NaN  

[5 rows x 55 columns]

In [18]:
# 欠損値も含めてユニークな値の個数をカウント
nunique_series_ver3 = df_cleaned_ver2.nunique(dropna=False)

# ユニークな値が1つのカラム名リスト
single_unique_cols_ver3 = nunique_series_ver3[nunique_series_ver3 == 2].index.tolist()

# カラム名とそのユニーク値を出力
for col in single_unique_cols_ver3:
    unique_val = df_cleaned_ver2[col].unique()
    print(f"{col}: {unique_val}")


Sensor: [nan 'Eye Tracker']
Presented Media width: [  nan 1920.]
Presented Media height: [  nan 1080.]
Presented Media position X (DACSpx): [nan  0.]
Presented Media position Y (DACSpx): [nan  0.]
Original Media width: [  nan 1920.]
Original Media height: [  nan 1080.]


### Sensor=Mouse以外に着目した後に要素が少ない特徴量について
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Mouse position X| マウス位置 X|`nan`|
|Mouse position Y| マウス位置 Y|`nan`|
|Sensor| センサー|`nan`, `Eye Tracker`|
|Presented Media width|提示されたメディアの幅|`nan` `1920`|
|Presented Media height|提示されたメディアの高さ|`nan`, `1080`|
|Presented Media position X (DACSpx)| 提示されたメディア位置 X|`nan`,  `0`|
|Presented Media position Y (DACSpx)| 提示されたメディアの位置 Y|`nan`,  `0`|
|Original Media width| オリジナルメディアの幅|`nan`, `1920`|
|Original Media height| オリジナルメディアの高さ|`nan`,`1080`|

In [19]:
df_sensor_eye

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
2               2                18888           843134751  Eye Tracker   
6               6                74431           843190294  Eye Tracker   
9               9                91074           843206937  Eye Tracker   
11             11               107761           843223624  Eye Tracker   
15             15               155926           843271789  Eye Tracker   
...           ...                  ...                 ...          ...   
11014       11014            198376477          1041492340  Eye Tracker   
11015       11015            198392520          1041508383  Eye Tracker   
11016       11016            198409704          1041525567  Eye Tracker   
11018       11018            198426396          1041542259  Eye Tracker   
11019       11019            198442443          1041558306  Eye Tracker   

       Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
2              8.584825e+08   NaN         NaN           NaN           NaN   
6              8.585381e+08   NaN         NaN           NaN           NaN   
9              8.585547e+08   NaN         NaN        1554.0        1004.0   
11             8.585714e+08   NaN         NaN           NaN           NaN   
15             8.586195e+08   NaN         NaN           NaN           NaN   
...                     ...   ...         ...           ...           ...   
11014          1.056828e+09   NaN         NaN         729.0         776.0   
11015          1.056844e+09   NaN         NaN         588.0        1003.0   
11016          1.056861e+09   NaN         NaN         550.0        1032.0   
11018          1.056878e+09   NaN         NaN         527.0        1034.0   
11019          1.056894e+09   NaN         NaN         528.0        1023.0   

       Gaze point left X  ...  Original Media height  Eye movement type  \
2                    NaN  ...                    NaN       EyesNotFound   
6                    NaN  ...                    NaN       EyesNotFound   
9                 1405.0  ...                    NaN       Unclassified   
11                   NaN  ...                    NaN       EyesNotFound   
15                   NaN  ...                    NaN       EyesNotFound   
...                  ...  ...                    ...                ...   
11014              714.0  ...                 1080.0            Saccade   
11015              604.0  ...                 1080.0            Saccade   
11016              561.0  ...                 1080.0            Saccade   
11018              579.0  ...                    NaN       Unclassified   
11019              559.0  ...                    NaN       Unclassified   

       Eye movement event duration  Eye movement type index  Fixation point X  \
2                             76.0                      1.0               NaN   
6                             76.0                      1.0               NaN   
9                             17.0                      1.0               NaN   
11                           171.0                      2.0               NaN   
15                           171.0                      2.0               NaN   
...                            ...                      ...               ...   
11014                         50.0                    654.0               NaN   
11015                         50.0                    654.0               NaN   
11016                         50.0                    654.0               NaN   
11018                         36.0                    104.0               NaN   
11019                         36.0                    104.0               NaN   

       Fixation point Y  Fixation point X (MCSnorm)  \
2                   NaN                         NaN   
6                   NaN                         NaN   
9                   NaN                         NaN   
11                  NaN                         NaN   
15                  NaN         

## 各問題毎のデータへ切り分け

In [20]:
# 結果を辞書に格納
extracted_data = {}

targets = df_filtered["Event value"].unique()
for target in targets:
    timestamps = df_filtered[df_filtered["Event value"] == target]["Recording timestamp"].iloc[0:2]
    if len(timestamps) < 2:
        continue  # start/endが不足している場合スキップ

    start, end = timestamps.iloc[0], timestamps.iloc[1]

    extracted = df_sensor_eye[
        (df_sensor_eye["Recording timestamp"] >= start) &
        (df_sensor_eye["Recording timestamp"] <= end)
    ].copy()

    extracted_data[target] = extracted  # 日本語のキーOK

In [21]:
extracted_data["視線誘導問題.002"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
8725        8725            160335143          1003451006  Eye Tracker   
8726        8726            160351186          1003467049  Eye Tracker   
8727        8727            160368425          1003484288  Eye Tracker   
8728        8728            160385066          1003500929  Eye Tracker   
8729        8729            160401109          1003516972  Eye Tracker   

      Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
8725          1.018789e+09   NaN         NaN         171.0         260.0   
8726          1.018805e+09   NaN         NaN         174.0         271.0   
8727          1.018822e+09   NaN         NaN         171.0         258.0   
8728          1.018839e+09   NaN         NaN         204.0         258.0   
8729          1.018855e+09   NaN         NaN         443.0         280.0   

      Gaze point left X  ...  Original Media height  Eye movement type  \
8725              168.0  ...                 1080.0           Fixation   
8726              166.0  ...                 1080.0           Fixation   
8727              166.0  ...                 1080.0           Fixation   
8728              206.0  ...                 1080.0            Saccade   
8729              439.0  ...                 1080.0            Saccade   

      Eye movement event duration  Eye movement type index  Fixation point X  \
8725                        267.0                    424.0             172.0   
8726                        267.0                    424.0             172.0   
8727                        267.0                    424.0             172.0   
8728                         66.0                    496.0               NaN   
8729                         66.0                    496.0               NaN   

      Fixation point Y  Fixation point X (MCSnorm)  \
8725             260.0                      0.0895   
8726             260.0                      0.0895   
8727             260.0                      0.0895   
8728               NaN                         NaN   
8729               NaN                         NaN   

      Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
8725                      0.2409               NaN               NaN  
8726                      0.2409               NaN               NaN  
8727                      0.2409               NaN               NaN  
8728                         NaN               NaN               NaN  
8729                         NaN               NaN               NaN  

[5 rows x 57 columns]

In [22]:
print(extracted_data["視線誘導問題.002"].nunique(dropna=False).to_string())

Unnamed: 0                             1988
Recording timestamp                    1988
Computer timestamp                     1988
Sensor                                    1
Eyetracker timestamp                   1988
Event                                     1
Event value                               1
Gaze point X                            979
Gaze point Y                            684
Gaze point left X                       998
Gaze point left Y                       672
Gaze point right X                      997
Gaze point right Y                      666
Gaze direction left X                  1910
Gaze direction left Y                  1874
Gaze direction left Z                  1585
Gaze direction right X                 1883
Gaze direction right Y                 1828
Gaze direction right Z                 1521
Pupil diameter left                     454
Pupil diameter right                    482
Pupil diameter filtered                 443
Validity left                   

In [23]:
extracted_data["視線誘導問題.001"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
2134        2134             51036896           894152759  Eye Tracker   
2135        2135             51053537           894169400  Eye Tracker   
2136        2136             51069580           894185443  Eye Tracker   
2137        2137             51086819           894202682  Eye Tracker   
2138        2138             51103460           894219323  Eye Tracker   

      Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
2134           909497321.0   NaN         NaN        1003.0         401.0   
2135           909513961.0   NaN         NaN        1010.0         403.0   
2136           909530003.0   NaN         NaN        1005.0         412.0   
2137           909547241.0   NaN         NaN        1006.0         406.0   
2138           909563881.0   NaN         NaN        1009.0         412.0   

      Gaze point left X  ...  Original Media height  Eye movement type  \
2134             1031.0  ...                 1080.0           Fixation   
2135             1025.0  ...                 1080.0           Fixation   
2136             1025.0  ...                 1080.0           Fixation   
2137             1026.0  ...                 1080.0           Fixation   
2138             1037.0  ...                 1080.0           Fixation   

      Eye movement event duration  Eye movement type index  Fixation point X  \
2134                        516.0                     63.0            1007.0   
2135                        516.0                     63.0            1007.0   
2136                        516.0                     63.0            1007.0   
2137                        516.0                     63.0            1007.0   
2138                        516.0                     63.0            1007.0   

      Fixation point Y  Fixation point X (MCSnorm)  \
2134             409.0                      0.5245   
2135             409.0                      0.5245   
2136             409.0                      0.5245   
2137             409.0                      0.5245   
2138             409.0                      0.5245   

      Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
2134                      0.3786               NaN               NaN  
2135                      0.3786               NaN               NaN  
2136                      0.3786               NaN               NaN  
2137                      0.3786               NaN               NaN  
2138                      0.3786               NaN               NaN  

[5 rows x 57 columns]

In [24]:
extracted_data["視線誘導問題.003"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
4210        4210             85433916           928549779  Eye Tracker   
4211        4211             85450525           928566388  Eye Tracker   
4212        4212             85466600           928582463  Eye Tracker   
4213        4213             85483839           928599702  Eye Tracker   
4214        4214             85500346           928616209  Eye Tracker   

      Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
4210           943892201.0   NaN         NaN         114.0         703.0   
4211           943908809.0   NaN         NaN         103.0         703.0   
4212           943924883.0   NaN         NaN         104.0         697.0   
4213           943942121.0   NaN         NaN         111.0         703.0   
4214           943958627.0   NaN         NaN         107.0         704.0   

      Gaze point left X  ...  Original Media height  Eye movement type  \
4210              104.0  ...                 1080.0           Fixation   
4211              106.0  ...                 1080.0           Fixation   
4212              104.0  ...                 1080.0           Fixation   
4213              106.0  ...                 1080.0           Fixation   
4214               97.0  ...                 1080.0           Fixation   

      Eye movement event duration  Eye movement type index  Fixation point X  \
4210                        217.0                    178.0             115.0   
4211                        217.0                    178.0             115.0   
4212                        217.0                    178.0             115.0   
4213                        217.0                    178.0             115.0   
4214                        217.0                    178.0             115.0   

      Fixation point Y  Fixation point X (MCSnorm)  \
4210             701.0                      0.0598   
4211             701.0                      0.0598   
4212             701.0                      0.0598   
4213             701.0                      0.0598   
4214             701.0                      0.0598   

      Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
4210                      0.6491               NaN               NaN  
4211                      0.6491               NaN               NaN  
4212                      0.6491               NaN               NaN  
4213                      0.6491               NaN               NaN  
4214                      0.6491               NaN               NaN  

[5 rows x 57 columns]

In [25]:
extracted_data["視線誘導問題.004"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
6439        6439            122376834           965492697  Eye Tracker   
6440        6440            122393434           965509297  Eye Tracker   
6441        6441            122409642           965525505  Eye Tracker   
6442        6442            122426706           965542569  Eye Tracker   
6443        6443            122443255           965559118  Eye Tracker   

      Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
6439           980832877.0   NaN         NaN         266.0         245.0   
6440           980849476.0   NaN         NaN         211.0         213.0   
6441           980865683.0   NaN         NaN         211.0         205.0   
6442           980882746.0   NaN         NaN         217.0         197.0   
6443           980899294.0   NaN         NaN         218.0         198.0   

      Gaze point left X  ...  Original Media height  Eye movement type  \
6439              255.0  ...                 1080.0            Saccade   
6440              206.0  ...                 1080.0            Saccade   
6441              215.0  ...                 1080.0           Fixation   
6442              216.0  ...                 1080.0           Fixation   
6443              220.0  ...                 1080.0           Fixation   

      Eye movement event duration  Eye movement type index  Fixation point X  \
6439                         50.0                    355.0               NaN   
6440                         50.0                    355.0               NaN   
6441                        233.0                    301.0             213.0   
6442                        233.0                    301.0             213.0   
6443                        233.0                    301.0             213.0   

      Fixation point Y  Fixation point X (MCSnorm)  \
6439               NaN                         NaN   
6440               NaN                         NaN   
6441             196.0                      0.1109   
6442             196.0                      0.1109   
6443             196.0                      0.1109   

      Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
6439                         NaN               NaN               NaN  
6440                         NaN               NaN               NaN  
6441                      0.1811               NaN               NaN  
6442                      0.1811               NaN               NaN  
6443                      0.1811               NaN               NaN  

[5 rows x 57 columns]

In [26]:
for key in extracted_data.keys():
    print(f"Key: {key}")
    df = extracted_data[key]
    nunique_series = df.nunique(dropna=False)
    single_unique_cols = nunique_series[nunique_series == 1].index.tolist()
    for col in single_unique_cols:
        unique_val = df[col].unique()
        print(f"  {col}: {unique_val}")

Key: 視線誘導問題.001
  Sensor: ['Eye Tracker']
  Event: [nan]
  Event value: [nan]
  Presented Stimulus name: ['視線誘導問題.001']
  Presented Media name: ['視線誘導問題.001.jpeg']
  Presented Media width: [1920.]
  Presented Media height: [1080.]
  Presented Media position X (DACSpx): [0.]
  Presented Media position Y (DACSpx): [0.]
  Original Media width: [1920.]
  Original Media height: [1080.]
  Mouse position X: [nan]
  Mouse position Y: [nan]
Key: 視線誘導問題.003
  Sensor: ['Eye Tracker']
  Event: [nan]
  Event value: [nan]
  Presented Stimulus name: ['視線誘導問題.003']
  Presented Media name: ['視線誘導問題.003.jpeg']
  Presented Media width: [1920.]
  Presented Media height: [1080.]
  Presented Media position X (DACSpx): [0.]
  Presented Media position Y (DACSpx): [0.]
  Original Media width: [1920.]
  Original Media height: [1080.]
  Mouse position X: [nan]
  Mouse position Y: [nan]
Key: 視線誘導問題.004
  Sensor: ['Eye Tracker']
  Event: [nan]
  Event value: [nan]
  Presented Stimulus name: ['視線誘導問題.004']
  Presen

In [27]:
# 空の辞書に結果を格納
cleaned_data = {}

for key, df in extracted_data.items():
    nunique_series = df.nunique(dropna=False)
    single_unique_cols = nunique_series[nunique_series == 1].index.tolist()

    # そのカラムを除外して新しいデータフレームを作成
    df_cleaned = df.drop(columns=single_unique_cols).copy()

    cleaned_data[key] = df_cleaned

    print(f"{key}: Removed {len(single_unique_cols)} columns -> Remaining {df_cleaned.shape[1]} columns")


視線誘導問題.001: Removed 13 columns -> Remaining 44 columns
視線誘導問題.003: Removed 13 columns -> Remaining 44 columns
視線誘導問題.004: Removed 13 columns -> Remaining 44 columns
視線誘導問題.002: Removed 13 columns -> Remaining 44 columns


### カラムの要素が1つしかない変数の説明 写真1
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-6`|
|Recording name| 録画名|`Recording7`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`16:43:00.114`|
|Recording start time UTC| 録画開始時刻(UTC) |`07:43:00.114`|
|Recording duration| 録画時間|`198601`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`11.9`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`0.87`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`41`pixels|
|Average validation accuracy | 平均バリデーション精度|`11.6`mm|
|Average validation precision SD | 平均バリデーション予測SD|`8.5`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`10.5`mm|
|Average validation accuracy |平均バリデーション精度| `0.87`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `0.64`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `0.79`degrees|
|Average validation accuracy |平均バリデーション精度| `40`pixels|
|Average validation precision SD |平均バリデーション予測SD| `30`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `37`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.001`|
|Presented Media name|提示されたメディア名| `視線誘導問題.001.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|

In [28]:
cleaned_data["視線誘導問題.001"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
2134        2134             51036896           894152759   
2135        2135             51053537           894169400   
2136        2136             51069580           894185443   
2137        2137             51086819           894202682   
2138        2138             51103460           894219323   
...          ...                  ...                 ...   
3899        3899             80341760           923457623   
3900        3900             80358401           923474264   
3901        3901             80374444           923490307   
3903        3903             80391683           923507546   
3904        3904             80408324           923524187   

      Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
2134           909497321.0        1003.0         401.0             1031.0   
2135           909513961.0        1010.0         403.0             1025.0   
2136           909530003.0        1005.0         412.0             1025.0   
2137           909547241.0        1006.0         406.0             1026.0   
2138           909563881.0        1009.0         412.0             1037.0   
...                    ...           ...           ...                ...   
3899           938800361.0         584.0         805.0              589.0   
3900           938817001.0         552.0         802.0              549.0   
3901           938833043.0         565.0         804.0              566.0   
3903           938850281.0         560.0         806.0              565.0   
3904           938866921.0         567.0         800.0              584.0   

      Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
2134              392.0               975.0               411.0  ...   
2135              416.0               996.0               390.0  ...   
2136              420.0               986.0               404.0  ...   
2137              404.0               987.0               409.0  ...   
2138              427.0               981.0               397.0  ...   
...                 ...                 ...                 ...  ...   
3899              795.0               579.0               814.0  ...   
3900              800.0               555.0               804.0  ...   
3901              771.0               565.0               836.0  ...   
3903              776.0               556.0               835.0  ...   
3904              779.0               549.0               821.0  ...   

      Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
2134                       0.3626                        0.5078   
2135                       0.3852                        0.5186   
2136                       0.3893                        0.5135   
2137                       0.3743                        0.5139   
2138                       0.3952                        0.5110   
...                           ...                           ...   
3899                       0.7365                        0.3016   
3900                       0.7407                        0.2891   
3901                       0.7140                        0.2941   
3903                       0.7184                        0.2894   
3904                       0.7214                        0.2858   

      Gaze point right Y (MCSnorm)  Eye movement type  \
2134                        0.3805           Fixation   
2135                        0.3611           Fixation   
2136                        0.3736           Fixation   
2137                        0.3784           Fixation   
2138                        0.3678           Fixation   
...                            ...                ...   
3899                        0.7539           Fixation   
3900                        0.7441           Fixation   
3901                        0.7742           Fixation   
3903                        0.7735           Fixation   
3904                        0.7601           Fixati

In [29]:
cleaned_data["視線誘導問題.001"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-6/user-6-pic-1.csv")

### カラムの要素が1つしかない変数の説明 写真4
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-6`|
|Recording name| 録画名|`Recording7`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`16:43:00.114`|
|Recording start time UTC| 録画開始時刻(UTC) |`07:43:00.114`|
|Recording duration| 録画時間|`198601`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`11.9`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`0.87`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`41`pixels|
|Average validation accuracy | 平均バリデーション精度|`11.6`mm|
|Average validation precision SD | 平均バリデーション予測SD|`8.5`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`10.5`mm|
|Average validation accuracy |平均バリデーション精度| `0.87`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `0.64`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `0.79`degrees|
|Average validation accuracy |平均バリデーション精度| `40`pixels|
|Average validation precision SD |平均バリデーション予測SD| `30`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `37`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.004`|
|Presented Media name|提示されたメディア名| `視線誘導問題.004.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|


In [30]:
cleaned_data["視線誘導問題.004"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
6439        6439            122376834           965492697   
6440        6440            122393434           965509297   
6441        6441            122409642           965525505   
6442        6442            122426706           965542569   
6443        6443            122443255           965559118   
...          ...                  ...                 ...   
8415        8415            155259039           998374902   
8416        8416            155276278           998392141   
8418        8418            155292919           998408782   
8419        8419            155308962           998424825   
8420        8420            155326201           998442064   

      Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
6439          9.808329e+08         266.0         245.0              255.0   
6440          9.808495e+08         211.0         213.0              206.0   
6441          9.808657e+08         211.0         205.0              215.0   
6442          9.808827e+08         217.0         197.0              216.0   
6443          9.808993e+08         218.0         198.0              220.0   
...                    ...           ...           ...                ...   
8415          1.013713e+09         962.0         596.0              971.0   
8416          1.013730e+09         959.0         614.0              959.0   
8418          1.013747e+09         966.0         620.0              966.0   
8419          1.013763e+09         971.0         612.0              983.0   
8420          1.013780e+09         972.0         613.0              984.0   

      Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
6439              229.0               278.0               261.0  ...   
6440              203.0               217.0               224.0  ...   
6441              207.0               206.0               204.0  ...   
6442              195.0               218.0               199.0  ...   
6443              183.0               216.0               213.0  ...   
...                 ...                 ...                 ...  ...   
8415              587.0               954.0               604.0  ...   
8416              611.0               958.0               617.0  ...   
8418              611.0               967.0               628.0  ...   
8419              597.0               958.0               627.0  ...   
8420              595.0               960.0               632.0  ...   

      Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
6439                       0.2120                        0.1446   
6440                       0.1878                        0.1128   
6441                       0.1913                        0.1075   
6442                       0.1802                        0.1136   
6443                       0.1696                        0.1124   
...                           ...                           ...   
8415                       0.5438                        0.4967   
8416                       0.5662                        0.4991   
8418                       0.5660                        0.5037   
8419                       0.5532                        0.4991   
8420                       0.5509                        0.4999   

      Gaze point right Y (MCSnorm)  Eye movement type  \
6439                        0.2413            Saccade   
6440                        0.2070            Saccade   
6441                        0.1888           Fixation   
6442                        0.1846           Fixation   
6443                        0.1969           Fixation   
...                            ...                ...   
8415                        0.5591           Fixation   
8416                        0.5708           Fixation   
8418                        0.5814           Fixation   
8419                        0.5802           Fixation   
8420                        0.5851           Fixati

In [31]:
cleaned_data["視線誘導問題.004"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-6/user-6-pic-4.csv")

### カラムの要素が1つしかない変数の説明 写真2
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-6`|
|Recording name| 録画名|`Recording7`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`16:43:00.114`|
|Recording start time UTC| 録画開始時刻(UTC) |`07:43:00.114`|
|Recording duration| 録画時間|`198601`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`11.9`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`0.87`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`41`pixels|
|Average validation accuracy | 平均バリデーション精度|`11.6`mm|
|Average validation precision SD | 平均バリデーション予測SD|`8.5`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`10.5`mm|
|Average validation accuracy |平均バリデーション精度| `0.87`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `0.64`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `0.79`degrees|
|Average validation accuracy |平均バリデーション精度| `40`pixels|
|Average validation precision SD |平均バリデーション予測SD| `30`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `37`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.002`|
|Presented Media name|提示されたメディア名| `視線誘導問題.002.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|

In [32]:
cleaned_data["視線誘導問題.002"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
8725         8725            160335143          1003451006   
8726         8726            160351186          1003467049   
8727         8727            160368425          1003484288   
8728         8728            160385066          1003500929   
8729         8729            160401109          1003516972   
...           ...                  ...                 ...   
10708       10708            193334248          1036450111   
10709       10709            193350291          1036466154   
10710       10710            193367530          1036483393   
10712       10712            193384171          1036500034   
10713       10713            193400214          1036516077   

       Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
8725           1.018789e+09         171.0         260.0              168.0   
8726           1.018805e+09         174.0         271.0              166.0   
8727           1.018822e+09         171.0         258.0              166.0   
8728           1.018839e+09         204.0         258.0              206.0   
8729           1.018855e+09         443.0         280.0              439.0   
...                     ...           ...           ...                ...   
10708          1.051786e+09        1639.0         456.0             1649.0   
10709          1.051802e+09        1642.0         453.0             1649.0   
10710          1.051819e+09        1645.0         518.0             1646.0   
10712          1.051836e+09        1609.0         689.0             1605.0   
10713          1.051852e+09        1607.0         685.0             1593.0   

       Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
8725               241.0               174.0               279.0  ...   
8726               251.0               182.0               291.0  ...   
8727               249.0               177.0               268.0  ...   
8728               254.0               203.0               263.0  ...   
8729               275.0               448.0               286.0  ...   
...                  ...                 ...                 ...  ...   
10708              461.0              1629.0               452.0  ...   
10709              466.0              1635.0               441.0  ...   
10710              495.0              1644.0               542.0  ...   
10712              693.0              1613.0               684.0  ...   
10713              680.0              1621.0               689.0  ...   

       Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
8725                        0.2228                        0.0906   
8726                        0.2327                        0.0946   
8727                        0.2301                        0.0921   
8728                        0.2349                        0.1055   
8729                        0.2547                        0.2333   
...                            ...                           ...   
10708                       0.4269                        0.8485   
10709                       0.4314                        0.8515   
10710                       0.4583                        0.8561   
10712                       0.6419                        0.8399   
10713                       0.6298                        0.8443   

       Gaze point right Y (MCSnorm)  Eye movement type  \
8725                         0.2579           Fixation   
8726                         0.2690           Fixation   
8727                         0.2483           Fixation   
8728                         0.2431            Saccade   
8729                         0.2646            Saccade   
...                             ...                ...   
10708                        0.4184           Fixation   
10709                        0.4082           Fixation   
10710                        0.5015            Saccade   
10712                        0.6338            Sacc

In [33]:
cleaned_data["視線誘導問題.002"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-6/user-6-pic-2.csv")

### カラムの要素が1つしかない変数の説明 写真3
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-6`|
|Recording name| 録画名|`Recording7`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`16:43:00.114`|
|Recording start time UTC| 録画開始時刻(UTC) |`07:43:00.114`|
|Recording duration| 録画時間|`198601`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`11.9`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`0.87`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`41`pixels|
|Average validation accuracy | 平均バリデーション精度|`11.6`mm|
|Average validation precision SD | 平均バリデーション予測SD|`8.5`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`10.5`mm|
|Average validation accuracy |平均バリデーション精度| `0.87`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `0.64`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `0.79`degrees|
|Average validation accuracy |平均バリデーション精度| `40`pixels|
|Average validation precision SD |平均バリデーション予測SD| `30`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `37`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.003`|
|Presented Media name|提示されたメディア名| `視線誘導問題.003.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|


In [34]:
cleaned_data["視線誘導問題.003"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
4210        4210             85433916           928549779   
4211        4211             85450525           928566388   
4212        4212             85466600           928582463   
4213        4213             85483839           928599702   
4214        4214             85500346           928616209   
...          ...                  ...                 ...   
6126        6126            117284622           960400485   
6129        6129            117301263           960417126   
6130        6130            117317490           960433353   
6132        6132            117334496           960450359   
6133        6133            117351183           960467046   

      Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
4210           943892201.0         114.0         703.0              104.0   
4211           943908809.0         103.0         703.0              106.0   
4212           943924883.0         104.0         697.0              104.0   
4213           943942121.0         111.0         703.0              106.0   
4214           943958627.0         107.0         704.0               97.0   
...                    ...           ...           ...                ...   
6126           975740977.0        1103.0         433.0             1100.0   
6129           975757617.0        1113.0         454.0             1108.0   
6130           975773843.0        1095.0         455.0             1086.0   
6132           975790847.0        1096.0         445.0             1079.0   
6133           975807533.0        1090.0         435.0             1082.0   

      Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
4210              695.0               123.0               711.0  ...   
4211              698.0               100.0               709.0  ...   
4212              685.0               104.0               709.0  ...   
4213              695.0               115.0               711.0  ...   
4214              684.0               117.0               725.0  ...   
...                 ...                 ...                 ...  ...   
6126              430.0              1106.0               437.0  ...   
6129              446.0              1118.0               461.0  ...   
6130              444.0              1104.0               466.0  ...   
6132              446.0              1113.0               444.0  ...   
6133              456.0              1098.0               415.0  ...   

      Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
4210                       0.6431                        0.0643   
4211                       0.6461                        0.0519   
4212                       0.6347                        0.0544   
4213                       0.6434                        0.0602   
4214                       0.6331                        0.0612   
...                           ...                           ...   
6126                       0.3984                        0.5762   
6129                       0.4131                        0.5821   
6130                       0.4108                        0.5748   
6132                       0.4129                        0.5797   
6133                       0.4223                        0.5716   

      Gaze point right Y (MCSnorm)  Eye movement type  \
4210                        0.6587           Fixation   
4211                        0.6561           Fixation   
4212                        0.6568           Fixation   
4213                        0.6581           Fixation   
4214                        0.6714           Fixation   
...                            ...                ...   
6126                        0.4042           Fixation   
6129                        0.4273           Fixation   
6130                        0.4313           Fixation   
6132                        0.4108           Fixation   
6133                        0.3839           Fixati

In [35]:
cleaned_data["視線誘導問題.003"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-6/user-6-pic-3.csv")